c7n/commands.py

```diff
@@ -249,10 +273,40 @@ def validate(options):
         log.error("Configuration invalid: {}".format(config_file))
         for e in errors:
             log.error("%s" % e)
+    if found_deprecations:
+        notes = footnotes()
+        if notes:
+            log.warning("deprecation footnotes:\n" + notes)
+        if options.check_deprecations == "error":
+            sys.exit(1)
     if errors:
         sys.exit(1)
 
 
+class SourceLocator:
```

Comment:
this feels like it should be in loader.py

Reply:
Hopefully we can remove this completely.

Reply:
Moved into loader.py and added a test.